### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by Luffytaro22**
> # Set up associations
> * Created `Classroom`, `Book`, and `Rental` classes
> * Created the `has-many/belongs-to` relationship between `Classroom` and `Student`
> * Created the `many-to-many` relationship between `Person` and `Book` using the intermediate class `Rental`

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  attr_accessor :label, :students
+
```
Comment (TadesseDev) [inline classroom.rb:8]: - [Optional] Dear @Luffytaro22, Its recommended to have your accessor and other helper methods at the top of the class, before the initializer method. Please consider following the good practice.

File: student.rb [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
     super(age, name, parent_permission: parent_permission)
     @classroom = classroom
+    classroom&.add_student(self)
   end
 
+  attr_accessor :classroom
+
   def play_hooky
     '¯(ツ)/¯'
   end
```
Comment (TadesseDev) [inline student.rb]: - [x] Dear @Luffytaro22, for the class Student, we need a method that serves as the 'belongs-to' side, `(a student belongs to a classroom)`, for that, please add a setter of a `classroom` that takes a `classroom` and assert the association is preserved. 

> Something like 

```ruby
@classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
```  

More can be found [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships)

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    @title = title
+    @author = author
+  end
+
+  attr_accessor :title, :author
+
+  # has-many relation with Rental
+  def rentals
+    Rental.all.select { |rental| rental.book == self }
+  end
```
Comment (TadesseDev) [inline book.rb]: - [x] dear @Luffytaro22, inside the book class, we need to initialize the rentals, as `@rentals = []`. Also once you add the rentals to the initializer please create `attr_accessor` to it. We need to be able to check the rentals for the book. 
- [x] Also, when we add a rental we need to create a new rental with the book as a self and add the newly created rental to the rentals list, This way the association will be preserved all the way. 

> The method should look something like...
```ruby
def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: person.rb [hunk 4]
```diff
@@ -1,4 +1,5 @@
 require_relative 'nameable'
+require_relative 'rental'
 
 class Person < Nameable
```
Comment (TadesseDev) [inline person.rb:4]: - [x] The same thing here, inside the `person` class, we need to initialize the rentals, as `@rentals = []`. Also once you add the rentals to the initializer please create `attr_accessor` to it. We need to be able to check the rentals for the person. 
- [x] Also, when we add a rental we need to create a new rental with the `person` as a self and add the newly created rental to the rentals list, This way the association will be preserved all the way. 

> The method should look something like...
```ruby
def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @Luffytaro22 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x] You have written a clean code ✅
 - [x]  Most of the requirement has been meet. ✅
 - [x]  Good PR title/description  ✅

 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by anagudelogu — APPROVED** [review]
# STATUS: APPROVED! 🍾 🎆 🥇 

Hi @Luffytaro22, 
I'm Andres.

You did a great job working on this milestone, your project is complete! 🚀 
There is nothing else to say other than... it's time to merge it 🤖  
Congratulations! 🎉 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**